backend/src/routes/UserProfile.ts

```diff
@@ -113,4 +116,141 @@ router.delete("/:id", async (req, res) => {
       .json({ error: `An error occurred while deleting the user profile with id: ${id}` });
   }
 });
+
+// Add a friend
+router.post("/add-friend", async (req, res) => {
+  const { friendId } = req.body;
+  const userId = userID;
+  try {
+    const friend = await prisma.friendship.create({
+      data: {
+        senderID: userId,
+        receiverID: friendId,
+      },
+    });
+    res.status(StatusCodes.CREATED).json(friend);
+  }
+  catch (error) {
+    console.log(error);
+    res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ error: "An error occurred while adding a friend." });
+  }
+});
+
+// view all friend requests
+router.get("/friend/requests", async (req, res) => {
+  const userId = userID;
+  try {
+    const friendRequests = await prisma.friendship.findMany({
+      where: {
+        receiverID: userId,
+        friendStatus: "PENDING",
+      },
+    });
+    res.status(StatusCodes.OK).json(friendRequests);
+  }
+  catch (error) {
+    console.log(error);
+    res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ error: "An error occurred while getting friend requests." });
+  }
+});
+
+// get all sent friend requests
+router.get("/friend/sent-requests", async (req, res) => {
```

Comment:
Can't this be combined with `GET /:userId/friend`? You can change the behaviour by passing a query parameter, `status=ACCEPTED` or `status=REJECTED`.

Reply:
Also the url here isn't consistent with "get all friends".